spec/notifications-spec.coffee

```diff
@@ -263,6 +263,28 @@ describe "Notifications", ->
 
           runs ->
             expect(atom.notifications.getNotifications().length).toBe 0
+      describe "when an exception contains newline", ->
+        beforeEach ->
+          stack = """
+            TypeError: undefined is not a function
+              at Object.module.exports.Pane.promptToSaveItem [as defaultSavePrompt] (/Applications/Atom.app/Contents/Resources/app/src/pane.js:490:23)
+              at Pane.promptToSaveItem (/Users/someguy/.atom/packages/save-session/lib/save-prompt.coffee:21:15)
+              at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:442:18)
+              at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app/node_modules/tabs/lib/tab-bar-view.js:174:22)
+              at space-pen-ul.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/archive-view/node_modules/atom-space-pen-views/node_modules/space-pen/vendor/jquery.js:4676:9)
+              at space-pen-ul.elemData.handle (/Applications/Atom.app/Contents/Resources/app/node_modules/archive-view/node_modules/atom-space-pen-views/node_modules/space-pen/vendor/jquery.js:4360:46)
+          """
+          detail = 'ok'
+
+          atom.notifications.addFatalError(stack, {detail, stack})
+          notificationContainer = workspaceElement.querySelector('atom-notifications')
+          fatalError = notificationContainer.querySelector('atom-notification.fatal')
+        it "title doesn't contain newline", ->
+          waitsForPromise ->
+            fatalError.getRenderPromise().then ->
+              issueTitle = fatalError.issue.getIssueTitle()
+          runs ->
+            expect(issueTitle).not.toContain "\n"
```

Comment:
I did some investigation and it seems like it does.  I tested this by forcing this to generate a url with the newlines not being removed and I get the following url to create a new issue:
```
https://github.com/atom/atom/issues/new?title=TypeError:+undefined+is+not+a+function%0A++at+Object.module.exports.Pane.promptToSaveItem+%5Bas+defau...&body=%5BEnter+steps+to+reproduce:%5D%0A%0A1.+...%0A2.+...%0A%0A**Atom**:+1.19.0-dev-85e65ffd1+x64%0A**Electron**:+1.6.9%0A**OS**:+Mac+OS+X+10.12.4%0A**Thrown+From**:+Atom+Core%0A%0A%0A%23%23%23+Stack+Trace%0A%0ATypeError:+undefined+is+not+a+function%0A++at+Object.module.exports.Pane.promptToSaveItem+%5Bas+defaultSavePrompt%5D+(/Applications/Atom.app/Contents/Resources/app/src/pane.js:490:23)%0A++at+PansdfsdfsdfsdPansdfsdfsdfsdfsdfsdffsdfsdfPansdfsdfsdfsdfsdfsdffsdfsdfPansdfsdfsdfsdfsdfsdffsdfsdffsdfsdffsdfsdf%0A%0A%0A%60%60%60%0AAt+undefined%0A%0ATypeError:+undefined+is+not+a+function%0A++at+Object.module.exports.Pane.promptToSaveItem+%5Bas+defaultSavePrompt%5D+(/Applications/Atom.app/Contents/Resources/app/src/pane.js:490:23)%0A++at+PansdfsdfsdfsdPansdfsdfsdfsdfsdfsdffsdfsdfPansdfsdfsdfsdfsdfsdffsdfsdfPansdfsdfsdfsdfsdfsdffsdfsdffsdfsdffsdfsdf%0A%0A%60%60%60%0A%0A%23%23%23+Commands%0A%0A%60%60%60%0A+++++-2:32.7.0+command-palette:toggle+(input.hidden-input)%0A+++++-2:31.8.0+editor:consolidate-selections+(input.hidden-input)%0A+++++-2:31.8.0+core:cancel+(input.hidden-input)%0A+++++-1:39.9.0+application:new-file+(ul.list-inline.tab-bar.inset-panel)%0A+++++-1:39+core:paste+(input.hidden-input)%0A+++++-1:28.6.0+core:copy+(input.hidden-input)%0A+++++-1:23.7.0+core:backspace+(input.hidden-input)%0A+++++-1:22.2.0+core:move-right+(input.hidden-input)%0A+++++-1:21.6.0+core:move-down+(input.hidden-input)%0A+++++-1:19.1.0+core:backspace+(input.hidden-input)%0A+++++-1:15.9.0+core:move-left+(input.hidden-input)%0A+++++-1:10.5.0+core:copy+(input.hidden-input)%0A%60%60%60%0A%0A%23%23%23+Non-Core+Packages%0A%0A%60%60%60%0Aatom-clock+0.1.7+%0Aatom-material-syntax+1.0.6+%0Aatom-material-ui+2.0.3+%0Amaterial-syntax+0.6.1+%0A%60%60%60%0A%0A
```

if you click on the link the title is:

```
TypeError: undefined is not a function  at Object.module.exports.Pane.promptToSaveItem [as defau...
```
Which has two spaces between function and at like you mention.  I'll add validate also to ensure we're getting the correct result from this.

Thanks,

Bryant